tests/conftest.py

```diff
@@ -24,11 +50,82 @@ def canonical_model() -> AmplitudeModel:
     return __create_model(formalism="helicity")
 
 
+@pytest.fixture(scope="session")
+def kinematics(helicity_model: AmplitudeModel) -> HelicityKinematics:
+    return HelicityKinematics.from_model(helicity_model)
+
+
+@pytest.fixture(scope="session")
+def phsp_sample(kinematics: HelicityKinematics) -> np.ndarray:
+    return generate_phsp(N_PHSP_EVENTS, kinematics, random_generator=RNG)
+
+
+@pytest.fixture(scope="session")
+def intensity(
+    helicity_model: AmplitudeModel,
+    kinematics: HelicityKinematics,
+    phsp_sample: np.ndarray,
+) -> IntensityTF:
+    # https://github.com/ComPWA/tensorwaves/issues/171
+    model = deepcopy(helicity_model)
+    builder = IntensityBuilder(model.particles, kinematics, phsp_sample)
+    return builder.create_intensity(model)
+
+
+@pytest.fixture(scope="session")
+def data_sample(
+    kinematics: HelicityKinematics,
+    intensity: IntensityTF,
+) -> np.ndarray:
+    return generate_data(
+        N_DATA_EVENTS, kinematics, intensity, random_generator=RNG
+    )
+
+
+@pytest.fixture(scope="session")
+def data_set(
+    kinematics: HelicityKinematics,
+    data_sample: np.ndarray,
+) -> dict:
+    return kinematics.convert(data_sample)
+
+
+@pytest.fixture(scope="session")
+def estimator(intensity: IntensityTF, data_set: dict) -> UnbinnedNLL:
+    return UnbinnedNLL(intensity, data_set)
+
+
+@pytest.fixture(scope="session")
+def free_parameters() -> dict:
+    return {
+        "Width_f(0)(500)": 0.3,
+        "Mass_f(0)(980)": 1,
+    }
+
+
+@pytest.fixture(scope="session")
+def fit_result(
+    estimator: UnbinnedNLL, free_parameters: dict, output_dir: str
+) -> dict:
+    optimizer = Minuit2(
+        callback=CallbackList(
+            [
+                CSVSummary(filename=output_dir + "fit_traceback.csv"),
+                YAMLSummary(filename=output_dir + "fit_result.yml"),
+            ]
+        )
+    )
+    return optimizer.optimize(estimator, free_parameters)
```

Comment:
Generally I don't like to have that many fixtures connected to one specific scenario, as this couples all of the tests together somewhat. For testing its ideal to also keep things decoupled so that we can quickly see which parts are going wrong. Concretely that would mean, that I would mock an Optimizer to test the callbacks. Once that is changed  the `free_parameters` fixture becomes redundant (can go directly to the Minuit2 test).

Generally we also have to start adding more edge cases to our tests, like an empty list of initial_parameters in the fit. That we expect some Error etc. Or a filename that already exists or an invalid path etc in the Callback tests.

Reply:
Better handled through #103